adafruit_emc2101/emc2101_lut.py

```diff
@@ -56,13 +56,42 @@ class FanSpeedLUT:
     """
 
     # 8 (Temperature, Speed) pairs in increasing order
-    _fan_lut = StructArray(EMC2101_Regs.LUT_BASE, "<B", 16)
+    _fan_lut = StructArray(emc2101_regs.LUT_BASE, "<B", 16)
+    _defer_update = 0
 
     def __init__(self, fan_obj):
+        self._defer_update = 0
         self.emc_fan = fan_obj
         self.lut_values = {}
         self.i2c_device = fan_obj.i2c_device
 
+    def __enter__(self):
+        """ 'with' wrapper: defer lut update until end of 'with'
+        so update_lut work can be done just once at the end of
+        setting the LUT.
+
+        Usage:
+        ```
+            with emc2101.lut as lut:
+                lut[20] = 0
+                lut[40] = 10
+        ```
+        """
+        # Use increment/decrement so nested with's are dealt with properly.
```

Comment:
Mainly the anti-nest code was because  I didn't want nesting to cause problems, rather than anything else. If python has a better solution I'm all ears. As far as I'm aware there's no use case for nested with's , but I wanted it to work properly even if some programmer did so.

Relating to the DigitalInOut case - personally the cost of preventing nesting being problematic is so low that I would always pay it and so would regard the case you mention as being a bug... but not going there now!

Reply:
It's not the `pyc` file size but the `.mpy` file size that microcontrollers use that I'm concerned about.  It sounds like things were already tight in another PR, so that is where the concern comes from.  Some of the microcontrollers are extremely tight on memory and storage.